File: Chapter1/1_2_CheckPermutation/check_permutation.py
def check_permutation(s: str, t: str) -> bool:
    """
    Assuming if they are of different length then we return false

    Time Complexity:
        O(n log n) because of sorting

    Space Complexity:
        O(n)
    """

    if len(s) != len(t):
        return False

    s = sorted(s.lower())
    t = sorted(t.lower())

    return s == t


print(check_permutation("hello", "ollhe"))
print(check_permutation("dallas", "taylor"))

File: Chapter1/1_2_CheckPermutation/checkPermutation.ts
const checkPermutation = (s: string, t: string): boolean => {
  /**
   * This is my solution
   *
   * time complexity: O(n log(n)) due to sorting
   * space complexity: O(n)
   *
   * Questions:
   * 1. is this case sensitive or not
   * 2. is whitespace significant
   */

  if (s.length != t.length) {
    return false;
  }

  s = s.toLocaleLowerCase().split("").sort().join("");
  t = t.toLocaleLowerCase().split("").sort().join("");
  return s === t;
};

console.log(checkPermutation("hello", "ollhe"));
console.log(checkPermutation("dallas", "taylor"));
